src/common/CryptoUtils.ts

```diff
@@ -132,6 +132,72 @@ export class CryptoUtils {
         };
     }
 
+    /**
+     * Converts a certificate from PEM to DER format.
+     *
+     * @param pemCertificate A string representing a certificate in PEM format.
+     * @returns A buffer containing the data of the certificate in DER format.
+     */
+    public static PEMtoDER(pemCertificate: string): Buffer {
+        const cert = this.getCertFromPEM(pemCertificate);
+        const derCertString = forge.asn1.toDer(forge.pki.certificateToAsn1(cert)).getBytes();
+        const derCertBuffer = Buffer.from(derCertString, 'binary');
+        return derCertBuffer;
+    }
+
+    /**
+     * Converts a certificate from DER to PEM format.
+     *
+     * @param derCertificate A buffer containing the binary data of the certificate in DER format.
+     * @returns A string representing a certificate in PEM format.
+     */
+    public static DERtoPEM(derCertificate: Buffer): string {
+        const cert = this.getCertFromDER(derCertificate);
+        const pemCert = forge.pki.certificateToPem(cert);
+        return pemCert;
+    }
+
+    /**
+     * Computes the subject hash (similar to subject_hash_old of OpenSSL). When invoking
+     * this method, the caller should either provide a DER or PEM certificate (but not both).
+     *
+     * @param derCertificate A buffer containing the data of the certificate in DER format.
+     * @param pemCertificate A string representing a certificate in PEM format.
+     * @returns A string representing the subject hash.
+     */
+    public static getSubjectHashOld(derCertificate?: Buffer, pemCertificate?: string): string {
+        if (!derCertificate && !pemCertificate) {
+            throw new SfError('Must provide either a DER or PEM certificate.');
+        } else if (derCertificate && pemCertificate) {
+            throw new SfError('Must provide either a DER or PEM certificate, but not both.');
+        }
+
+        const cert = derCertificate ? this.getCertFromDER(derCertificate) : this.getCertFromPEM(pemCertificate!);
```

Comment:
You could refactor lines 169-175 and 189-195 into one method?

Reply:
Sure but we first need to decide on the [above comment](https://github.com/forcedotcom/lwc-dev-mobile-core/pull/133/files#r1752904383), since that would dictate whether to even have these checks or not.